libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Extensions/RestClient.swift

```diff
@@ -113,6 +113,19 @@ extension RestClient {
         var records: [Record]?
     }
     
+    /// Sends a URLRequest
+    /// - Parameter urlRequest: The URLRequest to send
+    /// - Returns: A RestResponse containing the response data and metadata
+    /// - Throws: A RestClientError if the request fails
+    @objc(sendURLRequest:completion:)
+    public func send(urlRequest: URLRequest) async throws -> (Data, URLResponse) {
```

Comment:
Should we expose parameters for [networkServiceType](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/0b0eda4f92258530d4e3925d07873fc74f5ba828/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestRequest.h#L129) and [requiresAuthentication](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/0b0eda4f92258530d4e3925d07873fc74f5ba828/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestRequest.h#L188)?

Reply:
I'm actualy torn between two options. These are specific to a RestRequest, not a URLRequest. Therefore, I’m inclined to opt for the simpler approach and avoid exposing more details to maintain the API. However, I’m open to changing my mind if there’s a clear use case that I’m overlooking.

Reply:
Could we add them as optional parameters on sendRequest ?
```swift
@objc(sendURLRequest:completion:)
public func send(
    urlRequest: URLRequest,
    networkServiceType: RestRequest.NetWorkServiceType = .default,
    requiresAuthentication: Bool = true
) async throws -> (Data, URLResponse) {
...
}
```

Could we also fix the comment for https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/dev/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestRequest.h#L41C1-L49C49 (it's copied over from SFRestMethod.

Also, it's unfortunate that the swift name for network service type is NetWorkServiceType but we can't fix that :-(

Reply:
I see. Hopefully there is nobody out there that want to use the new API from objective-c code.